bigtable-client-core-parent/bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableSession.java

```diff
@@ -230,16 +232,19 @@ public BigtableSession(BigtableOptions opts) throws IOException {
     RetryOptions retryOptions = options.getRetryOptions();
     CredentialOptions credentialOptions = options.getCredentialOptions();
     try {
-      ClientInterceptor headerInterceptor =
+      ClientInterceptor credentialsInterceptor =
           credentialsCache.getCredentialsInterceptor(credentialOptions, retryOptions);
-      if (headerInterceptor != null) {
-        headerInterceptorBuilder.add(headerInterceptor);
+      if (credentialsInterceptor != null) {
+        clientInterceptorsList.add(credentialsInterceptor);
       }
     } catch (GeneralSecurityException e) {
       throw new IOException("Could not initialize credentials.", e);
     }
 
-    headerInterceptors = headerInterceptorBuilder.build().toArray(new ClientInterceptor[0]);
+    clientInterceptorsList.add(throttlingClientInterceptor);
```

Comment:
I think we should throttle before fetching the oauth token. If the token is about to expire and we throttle the request then when we unblock the request, the token can be expired.